### ISSUE TITLE
[Bug] Dropout 未使用可控 RNG，導致訓練不可重現

### ISSUE THREAD

**Issue by s9213712** [s9213712/minicnn#35]
## 問題描述

`src/minicnn/ops/nn_ops.py` 中 `dropout()` 使用全域 `np.random.random()`：

```python
def dropout(x: Tensor, p: float = 0.5, training: bool = True) -> Tensor:
    if not training or p == 0.0:
        return x
    mask = (np.random.random(x.data.shape) > p).astype(np.float32) / (1.0 - p)
```

這使用的是 NumPy 全域隨機狀態，沒有任何機制可以設定 seed 使訓練可重現。

## 影響

- **嚴重性**: Major（科學實驗可重現性）
- 相同的訓練配置、相同的資料，每次運行結果不同
- 無法復現已記錄的訓練結果
- 與 PyTorch 的 `torch.manual_seed()` 行為不一致

## 建議修復

在訓練開始時允許設定 seed：

```python
# 方案 A：全域 seed 設定
import numpy as np
def set_random_seed(seed: int):
    np.random.seed(seed)

# 方案 B：使用 Generator 物件（推薦）
_rng = np.random.default_rng()  # 模組級別

def dropout(x, p=0.5, training=True, rng=None):
    rng = rng or _rng
    mask = (rng.random(x.data.shape) > p).astype(np.float32) / (1.0 - p)
```

## 相關位置

- `src/minicnn/ops/nn_ops.py:50-62` — `dropout()` 函數
- `src/minicnn/training/loop.py` — 訓練迴圈，應在此設定 seed
